presto-main/src/main/java/com/facebook/presto/execution/CreateMaterializedViewTask.java

```diff
@@ -0,0 +1,153 @@
+import com.facebook.presto.sql.analyzer.Analyzer;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.tree.CreateMaterializedView;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.transaction.TransactionManager;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.createQualifiedObjectName;
+import static com.facebook.presto.metadata.MetadataUtil.toSchemaTableName;
+import static com.facebook.presto.spi.StandardErrorCode.ALREADY_EXISTS;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_FOUND;
+import static com.facebook.presto.sql.NodeUtils.mapFromProperties;
+import static com.facebook.presto.sql.SqlFormatterUtil.getFormattedSql;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.MATERIALIZED_VIEW_ALREADY_EXISTS;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.NOT_SUPPORTED;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+import static java.util.Objects.requireNonNull;
+
+public class CreateMaterializedViewTask
+        implements DataDefinitionTask<CreateMaterializedView>
+{
+    private final SqlParser sqlParser;
+
+    @Inject
+    public CreateMaterializedViewTask(SqlParser sqlParser)
+    {
+        this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+    }
+    @Override
+    public String getName()
+    {
+        return "CREATE MATERIALIZED VIEW";
+    }
+
+    @Override
+    public ListenableFuture<?> execute(CreateMaterializedView statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName viewName = createQualifiedObjectName(session, statement, statement.getName());
+
+        Optional<TableHandle> viewHandle = metadata.getTableHandle(session, viewName);
+        if (viewHandle.isPresent()) {
+            if (!statement.isNotExists()) {
+                throw new SemanticException(MATERIALIZED_VIEW_ALREADY_EXISTS, statement, "Materialized view '%s' already exists", viewName);
+            }
+            return immediateFuture(null);
+        }
+
+        // Relaxed acl for the moment and use create table acl.
+        accessControl.checkCanCreateTable(session.getRequiredTransactionId(), session.getIdentity(), session.getAccessControlContext(), viewName);
+
+        Analyzer analyzer = new Analyzer(session, metadata, sqlParser, accessControl, Optional.empty(), parameters, stateMachine.getWarningCollector());
+        Analysis analysis = analyzer.analyze(statement);
```

Comment:
There are a lot of checks we should do for `analysis`. These checks will also be applied to read path. For example, we should only allow simple group by queries at the moment.